src/control/modify.js

```diff
@@ -43,71 +43,174 @@ class ModifyControl extends Control {
-        const featureStyles = styles.slice(0, styles.indexOf(selectStyles[0]));
-        evt.element.setStyle(featureStyles);
-      });
-    }
+    this.selectStyle = options.style;
 
     /**
      * @type {ol.interaction.Modify}
      * @private
      */
     this.modifyInteraction = new ol.interaction.Modify({
-      features: this.selectInteraction.getFeatures(),
+      source: this.source,
       style: options.modifyStyle,
     });
+
+    /**
+     * @type {ol.interaction.Pointer}
+     * @private
+     */
+    this.moveInteraction = new ol.interaction.Pointer({
+      handleDownEvent: this.startMoveFeature.bind(this),
+      handleDragEvent: this.moveFeature.bind(this),
+      handleUpEvent: this.stopMoveFeature.bind(this),
+      handleMoveEvent: this.selectFeature.bind(this),
+    });
+  }
+
+  /**
+   * Handle the event of the delete event listener.
+   * @param {Event} evt Event.
+   * @private
+   */
+  deleteFeature(evt) {
+    if (evt.key === 'Delete' && this.feature) {
+      this.source.removeFeature(this.feature);
+      this.feature = null;
+    }
+  }
+
+  /**
+   * Handle the down event of the move interaction.
+   * @param {ol.MapBrowserEvent} evt Event.
+   * @private
+   */
+  startMoveFeature(evt) {
+    if (this.feature && this.modifyActive === false) {
+      if (this.feature.getGeometry() instanceof ol.geom.Point) {
+        const extent = this.feature.getGeometry().getExtent();
+        this.coordinate = ol.extent.getCenter(extent);
+      } else {
+        this.coordinate = evt.coordinate;
+      }
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Handle the drag event of the move interaction.
+   * @param {ol.MapBrowserEvent} evt Event.
+   * @private
+   */
+  moveFeature(evt) {
+    if (this.modifyActive === false) {
+      const deltaX = evt.coordinate[0] - this.coordinate[0];
+      const deltaY = evt.coordinate[1] - this.coordinate[1];
+
+      this.feature.getGeometry().translate(deltaX, deltaY);
+      this.coordinate = evt.coordinate;
+    }
+  }
+
+  /**
+   * Handle the up event of the pointer interaction.
+   * @param {ol.MapBrowserEvent} evt Event.
+   * @private
+   */
+  stopMoveFeature() {
+    this.coordinate = null;
+    this.feature = null;
+    this.editor.setEditFeature(null);
+    return false;
+  }
+
+  /**
+   * Handle the move event of the move interaction.
+   * @param {ol.MapBrowserEvent} evt Event.
+   * @private
+   */
+  selectFeature(evt) {
+    if (this.feature) {
+      // Remove the select style dynamically when the feature had its own style.
+      if (this.feature.getStyleFunction()) {
+        const styles = getStyles(this.feature.getStyleFunction(), null);
+        const selectStyles = getStyles(this.selectStyle, this.feature);
+        this.feature.setStyle(styles.slice(0, styles.indexOf(selectStyles[0])));
+      }
+    }
+
+    this.feature = evt.map.forEachFeatureAtPixel(evt.pixel, (f) => {
+      if (this.source.getFeatures().indexOf(f) > -1) {
```

Comment:
To avoid going through all features of all layers, you could use the [layerFilter options](https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html#forEachFeatureAtPixel).